cpp/game_main.cpp

```diff
@@ -241,6 +242,74 @@ GameMain::GameMain(Engine *engine)
 	delete teamcolor_frag;
 	delete alphamask_vert;
 	delete alphamask_frag;
+
+	// initialize global keybinds
+	auto &global_keybind_context = engine->get_keybind_manager().get_global_keybind_context();
+
+	global_keybind_context.bind(keybinds::action_t::STOP_GAME, [this]() {
+		this->engine->stop();
+	});
+	global_keybind_context.bind(keybinds::action_t::TOGGLE_HUD, [this]() {
+		this->engine->drawing_huds = !this->engine->drawing_huds;
+	});
+	global_keybind_context.bind(keybinds::action_t::SCREENSHOT, [this]() {
+		this->engine->get_screenshot_manager().save_screenshot();
+	});
+	global_keybind_context.bind(keybinds::action_t::TOGGLE_DEBUG_OVERLAY, [this]() {
+		this->engine->drawing_debug_overlay = !this->engine->drawing_debug_overlay;
+	});
+	global_keybind_context.bind(keybinds::action_t::TOGGLE_DEBUG_GRID, [this]() {
+		this->debug_grid_active = !this->debug_grid_active;
+	});
+	global_keybind_context.bind(keybinds::action_t::QUICK_SAVE, [this]() {
+		gameio::save(this, "default_save.txt");
+	});
+	global_keybind_context.bind(keybinds::action_t::QUICK_LOAD, [this]() {
+		gameio::load(this, "default_save.txt");
+	});
+	global_keybind_context.bind(keybinds::action_t::TOGGLE_PROFILER, [this]() {
+		if (this->external_profiler.currently_profiling) {
+			this->external_profiler.stop();
+			this->external_profiler.show_results();
+		} else {
+			this->external_profiler.start();
+		}
+	});
+
+	// Local keybinds
+	keybind_context.bind(keybinds::action_t::TOGGLE_BLENDING, [this]() {
+		this->terrain->blending_enabled = !terrain->blending_enabled;
+	});
+	keybind_context.bind(keybinds::action_t::TOGGLE_CONSTRUCT_MODE, [this]() {
+		this->construct_mode = !this->construct_mode;
+	});
+	keybind_context.bind(keybinds::action_t::TOGGLE_UNIT_DEBUG, [this]() {
+		UnitAction::show_debug = !UnitAction::show_debug;
+	});
+	keybind_context.bind(keybinds::action_t::TRAIN_OBJECT, [this]() {
+		// attempt to train editor selected object
+		if ( this->datamanager.producer_count() > 0 ) {
+			UnitProducer *producer = this->datamanager.get_producer_index(this->editor_current_building);
+			Command cmd(this->players[0], producer);
+			this->selection.all_invoke(cmd);
+		}
+	});
+	keybind_context.bind(keybinds::action_t::ENABLE_BUILDING_PLACEMENT, [this]() {
+		this->building_placement = true;
+	});
+	keybind_context.bind(keybinds::action_t::DISABLE_SET_ABILITY, [this]() {
+		this->use_set_ability = false;
+	});
+	keybind_context.bind(keybinds::action_t::SET_ABILITY_MOVE, [this]() {
+		this->use_set_ability = true;
+		this->ability = ability_type::move;
+	});
+	keybind_context.bind(keybinds::action_t::SET_ABILITY_GATHER, [this]() {
+		this->use_set_ability = true;
+		this->ability = ability_type::gather;
+	});
```

Comment:
can you add `this->keybind_context` pls? It's visible directly that it's a member variable that way.